lib/webhook-proxy.js

```diff
@@ -1,14 +1,31 @@
 module.exports = ({url, port, path, logger}) => {
   try {
+    var goodUrl = true // url is considered to be true
     const SmeeClient = require('smee-client')
-
-    const smee = new SmeeClient({
-      source: url,
-      target: `http://localhost:${port}${path}`,
-      logger
-    })
-    return smee.start()
+    if (!validURL(url)) {
+      goodUrl = false // url is invalid
+      const error = 'The provided Webhook Proxy URL is invalid.'
+      throw error
+    } else {
+      const smee = new SmeeClient({
+        source: url,
+        target: `http://localhost:${port}${path}`,
+        logger
+      })
+      return smee.start()
+    }
   } catch (err) {
-    logger.warn('Run `npm install --save-dev smee-client` to proxy webhooks to localhost.')
+    if (goodUrl) logger.warn('Run `npm install --save-dev smee-client` to proxy webhooks to localhost')
+    else logger.warn(err)
   }
 }
+// function to validate the url (taken from stack-overflow)
+function validURL (str) {
```

Comment:
I know that this function is hoisted, but I'm personally really not a fan of hoisting (it can cause unexpected errors) - can you move this function declaration to the top of the file?